src/ccloudStatus/api.ts

```diff
@@ -0,0 +1,23 @@
+import { logError } from "../errors";
+import { CCloudStatusSummary, CCloudStatusSummaryFromJSON } from "./types";
+
+const CCLOUD_STATUS_API_URL = "https://status.confluent.cloud/api/v2/summary.json";
+
+/** Fetches the current Confluent Cloud status from the public Statuspage Status API. */
+export async function fetchCCloudStatus(): Promise<CCloudStatusSummary | undefined> {
```

Comment:
Considered having both non-200 results as well as errors caught here being presented as raised errors, therefore no need to ever return undefined, then simplifying `refreshCCloudStatus()` an iota?

Also, is the non-200 manual error raised here ... then immediately caught and discarded and undefined returned? That seems byzantine. +1 for anything other than a 200 response end up throwing, letting the throw break that poller update cycle, then to be retried in 2 minutes no problem.